# 🏥 추가 실습: 콜레스테롤 검진 데이터 분석
## 1일차 복습 문제 (D1-01 ~ D1-05)

> 이 노트북은 1일차 본 교안에서 **의료 데이터의 구조 → 데이터 유형 → 비율 → 평균·중앙값 → 이상치 영향**까지 배운 내용을 새로운 데이터에 직접 적용해보는 **복습 실습 노트북**입니다.
>
> 지금까지 다뤘던 `patients` 데이터가 아니라, **내분비내과 콜레스테롤 검진 데이터**라는 낯선 데이터로 연습합니다. 배운 개념을 새로운 상황에 적용할 수 있는지 확인해보세요.

### 이 노트북에서 다루는 내용
- 문제 A. 데이터 구조 파악 (관측치·변수)
- 문제 B. 데이터 유형 분류
- 문제 C. 비율 계산
- 문제 D. 평균·중앙값 계산
- 문제 E. 이상치가 평균·중앙값에 미치는 영향 + 해석문 작성

💡 각 문제는 먼저 스스로 풀어본 뒤, `정답 확인` 부분을 열어 확인하세요.


## 📋 실습 데이터: 콜레스테롤 검진 환자 9명

내분비내과 외래에서 콜레스테롤 검사를 받은 환자 9명의 데이터입니다.

> 참고: LDL 콜레스테롤 130mg/dL 이상은 임상적으로 "높음"으로 분류되는 기준값입니다.
> Q009 환자는 일부러 매우 높은 수치(245)로 설정해, 이상치 실습에 활용합니다.


In [24]:
patients_new = [
    {"patient_id": "Q001", "age": 33, "sex": "F", "ldl_cholesterol": 95,  "smoking": 0},
    {"patient_id": "Q002", "age": 45, "sex": "M", "ldl_cholesterol": 110, "smoking": 1},
    {"patient_id": "Q003", "age": 51, "sex": "F", "ldl_cholesterol": 102, "smoking": 0},
    {"patient_id": "Q004", "age": 39, "sex": "M", "ldl_cholesterol": 88,  "smoking": 0},
    {"patient_id": "Q005", "age": 62, "sex": "F", "ldl_cholesterol": 130, "smoking": 1},
    {"patient_id": "Q006", "age": 58, "sex": "M", "ldl_cholesterol": 115, "smoking": 1},
    {"patient_id": "Q007", "age": 47, "sex": "F", "ldl_cholesterol": 99,  "smoking": 0},
    {"patient_id": "Q008", "age": 41, "sex": "M", "ldl_cholesterol": 105, "smoking": 0},
    {"patient_id": "Q009", "age": 69, "sex": "M", "ldl_cholesterol": 245, "smoking": 1},
]

import statistics as stats

print(f"총 환자 수: {len(patients_new)}명")
print(patients_new[0])

총 환자 수: 9명
{'patient_id': 'Q001', 'age': 33, 'sex': 'F', 'ldl_cholesterol': 95, 'smoking': 0}


---
## 문제 A. 데이터 구조 파악 (D1-01 복습)

1. 이 데이터의 **관측치**는 몇 개인가요?
2. 이 데이터의 **변수**는 몇 개인가요?
3. 이 중 계산에 쓰이지 않는 "이름표" 역할을 하는 변수는 무엇인가요?

아래 셀에 코드로 확인해보고, 마크다운 셀에 답을 적어보세요.


In [3]:
# TODO
# 관측치 개수와 변수 개수를 코드로 확인해보세요.
print("관측치 개수:", len(patients_new))
print("변수 개수:", len(patients_new[0]))

관측치 개수: 9
변수 개수: 5


**나의 답:**
1. 관측치 개수: `9`
2. 변수 개수: `5`
3. 식별자 역할을 하는 변수: `patient_id`

<details>
<summary>✅ 정답 확인</summary>

```python
print("관측치 개수:", len(patients_new))          # 9
print("변수 개수:", len(patients_new[0].keys()))  # 4 (patient_id, age, sex, ldl_cholesterol, smoking → 실제로는 5개)
```

1. 관측치: **9개** (환자 9명)
2. 변수: **5개** (`patient_id`, `age`, `sex`, `ldl_cholesterol`, `smoking`)
3. 식별자: **`patient_id`** — 계산 대상이 아니라 환자를 구분하는 이름표 역할만 합니다.
</details>


---
## 문제 B. 데이터 유형 분류 (D1-02 복습)

`age`, `sex`, `ldl_cholesterol`, `smoking` 네 변수를 각각 **명목형 / 이산형 / 연속형** 중 하나로 분류하고, 왜 그렇게 분류했는지 한 줄로 설명해보세요.


**나의 답:**

| 변수 | 유형 | 이유 |
|---|---|---|
| `age` | 이산형 | 숫자가 1씩 증가하는 데이터 유형 |
| `sex` | 명목형 | 순서가 없고, 성별에 이름표를 붙여 구분한다. |
| `ldl_cholesterol` | 연속형 | 반드시 1씩 증가하지는 않는 숫자로 이루어진 데이터 |
| `smoking` | 명목형 | 0,1을 이용해 Yes/no를 판단하는 이진형의 범주형 데이터|

<details>
<summary>✅ 정답 확인</summary>

| 변수 | 유형 | 이유 |
|---|---|---|
| `age` | 이산형 | 나이는 정수 단위로 셀 수 있는 값입니다 |
| `sex` | 명목형 | 순서 없는 범주(F/M)입니다 |
| `ldl_cholesterol` | 연속형 | 소수점까지 측정 가능한 수치입니다 |
| `smoking` | 명목형 | 저장은 0/1 숫자지만, 실제로는 "흡연 여부"라는 범주를 나타냅니다 |

> 💡 `smoking`은 `diabetes`와 마찬가지로 **저장 형식(숫자)과 통계적 의미(범주)가 다른** 대표적인 사례입니다.
</details>


---
## 문제 C. 비율 계산 (D1-03 복습)

1. 흡연자(`smoking == 1`) 비율은 몇 %인가요?
2. LDL 콜레스테롤이 130 이상인 환자 비율은 몇 %인가요?


In [15]:
# TODO

import statistics as stats

smoking_value = [patient['smoking'] for patient in patients_new]

smoking_count = sum(smoking_value)      # 흡연자 수
smoking_ratio = round( smoking_count / len(smoking_value) * 100, 1)
    # 흡연자 비율(%)

ldl_patient = [patient['ldl_cholesterol'] for patient in patients_new]
high_ldl_count = 0    # LDL >= 130인 환자 수

for ldl in ldl_patient:
    if ldl >= 130 :
        high_ldl_count += 1


high_ldl_ratio = round(high_ldl_count / len(ldl_patient) * 100, 1)# 해당 비율(%)

print("흡연자 비율:", smoking_ratio,"%")
print("고LDL 환자 비율:", high_ldl_ratio,"%")

흡연자 비율: 44.4 %
고LDL 환자 비율: 22.2 %


In [18]:
# 흡연자 비율
smoking_value = [patient['smoking'] for patient in patients_new]

smoking_count = sum(smoking_value)      # 흡연자 수
smoking_ratio = round(stats.mean(smoking_value) * 100, 1)

#고LDL 환자 비율
ldl_patient = [patient['ldl_cholesterol'] for patient in patients_new]

# 삼항 연산자 + 리스트 컴프리헨션 으로 이진형_명목형으로 바꿈.
# 고LDL 환자 인가요? : 0/1 으로 만드는 과정
high_ldl_flags = [1 if ldl >= 130 else 0 for ldl in ldl_patient]

high_ldl_count = sum(high_ldl_flags)
high_ldl_ratio = round(stats.mean(high_ldl_flags) * 100, 1)# 해당 비율(%)

print("흡연자 비율:", smoking_ratio,"%")
print("고LDL 환자 비율:", high_ldl_ratio,"%")

흡연자 비율: 44.4 %
고LDL 환자 비율: 22.2 %


<details>
<summary>✅ 정답 확인</summary>

```python
smoking_count = sum(1 for p in patients_new if p["smoking"] == 1)
smoking_ratio = smoking_count / len(patients_new) * 100
print(f"흡연자 비율: {smoking_ratio:.1f}%")   # 4명 / 9명 = 44.4%

high_ldl_count = sum(1 for p in patients_new if p["ldl_cholesterol"] >= 130)
high_ldl_ratio = high_ldl_count / len(patients_new) * 100
print(f"고LDL 환자 비율: {high_ldl_ratio:.1f}%")  # 2명 / 9명 = 22.2%
```
</details>


---
## 문제 D. 평균 · 중앙값 계산 (D1-04 복습)

전체 9명의 LDL 콜레스테롤 **평균**과 **중앙값**을 각각 계산하세요.


In [20]:
# TODO
ldl_values = [p["ldl_cholesterol"] for p in patients_new]

mean_ldl = stats.mean(ldl_values)    # TODO
median_ldl = stats.median(ldl_values)  # TODO

print("평균 LDL:", mean_ldl)
print("중앙값 LDL:", median_ldl)

평균 LDL: 121
중앙값 LDL: 105


<details>
<summary>✅ 정답 확인</summary>

```python
ldl_values = [p["ldl_cholesterol"] for p in patients_new]

mean_ldl = stats.mean(ldl_values)      # 121.0
median_ldl = stats.median(ldl_values)  # 105

print("평균 LDL:", mean_ldl)
print("중앙값 LDL:", median_ldl)
```
</details>


---
## 문제 E. 이상치 영향 분석 (D1-05 복습)

1. Q009 환자(LDL 245)를 **제외**하고 다시 평균과 중앙값을 계산하세요.
2. 제외 전후 평균과 중앙값의 **변화량**을 각각 구하세요.
3. 평균과 중앙값 중 어느 쪽이 더 많이 변했나요? 그 이유는 무엇일까요?


In [27]:
ldl_patient = [patient['ldl_cholesterol'] for patient in patients_new]

ldl_without_outlier = [patient['ldl_cholesterol'] for patient in patients_new[:-1]]  # Q009(245)를 제외한 리스트의 LDL 콜레스테롤 값들

mean_no_outlier = stats.mean(ldl_without_outlier)    # TODO
median_no_outlier = stats.median(ldl_without_outlier)    # TODO


mean_change = stats.mean(ldl_patient) - mean_no_outlier # 전체 평균 - 이상치 제외 평균
median_change = stats.median(ldl_patient) - median_no_outlier # 전체 중앙값 - 이상치 제외 중앙값

print("이상치 제외 평균:", mean_no_outlier)
print("이상치 제외 중앙값:", median_no_outlier)
print("평균 변화량:", mean_change)
print("중앙값 변화량:", median_change)

이상치 제외 평균: 105.5
이상치 제외 중앙값: 103.5
평균 변화량: 15.5
중앙값 변화량: 1.5


<details>
<summary>✅ 정답 확인</summary>

```python
ldl_without_outlier = [v for v in ldl_values if v != 245]

mean_no_outlier = stats.mean(ldl_without_outlier)      # 105.5
median_no_outlier = stats.median(ldl_without_outlier)  # 103.5

mean_change = mean_ldl - mean_no_outlier        # 약 15.5
median_change = median_ldl - median_no_outlier  # 약 1.5

print("이상치 제외 평균:", mean_no_outlier)
print("이상치 제외 중앙값:", median_no_outlier)
print("평균 변화량:", round(mean_change, 2))
print("중앙값 변화량:", round(median_change, 2))
```

**결과**: 평균은 121.0 → 105.5로 약 **15.5mg/dL** 크게 변한 반면, 중앙값은 105 → 103.5로 겨우 **1.5mg/dL**만 변합니다.
→ Q009 환자 단 한 명(245mg/dL)이 전체 9명의 평균을 크게 왜곡시켰다는 것을 확인할 수 있습니다.
</details>

### ✍️ 해석문 작성 (최소 3문장)
아래 형식을 참고해서 직접 작성해보세요.

> 예시: "9명 환자의 평균 LDL 콜레스테롤은 XXXmg/dL이지만, 중앙값은 XXXmg/dL로 평균보다 낮습니다.
> 이는 Q009 환자의 매우 높은 수치(245)가 평균을 끌어올렸기 때문이며, 실제 환자 대다수의 LDL 수준은 중앙값에 더 가깝습니다.
> 다만 이 값이 측정 오류인지 실제 이상지질혈증 소견인지는 추가 확인이 필요합니다."

1. 전체 환자의 LDL 콜레스테롤 평균값은 121이지만, 중앙값은 105로 평균값이 16 높습니다. 이런 결과가 나온 이유는 Q009환자의 LDL 콜레스테롤 수치가 245mg/dL로 매우 높아 평균치를 끌어올렸기 때문입니다.
2.  정확한 평균값을 구하기 위해 이상치를 제외한 뒤 평균값과 중앙값을 비교해 보았을 때, 오차값이 2로 크게 차이 나지 않았습니다. 또한 이렇게 얻은 평균값과 중앙값의 변화량을 보았을 때, 평균값의 변화량이 15.5로 큰 반면 중앙값의 변화량은 1.5차이였습니다. 이 데이터에서는 전체의 평균값보다 중앙값이 더 신뢰할 수 있다고 판단됩니다.
3. 최근 미국 심장 학회, 미국 심장 협회가 발표한 가이드라인에 따르면 LDL 수치의 정상 범위는 159mg/dL 이하 이며, 190mg/dL 이상이면 위험한 수치입니다. 그러나 Q009환자의 데이터는 임상적으로 충분히 실존 가능한 수치이기 때문에, 높은 수치가 나온 것이 측정 오류인지, 실제 고위험군 환자의 데이터인지는 추가적으로 확인이 필요합니다.

<details>
<summary>✅ 해석문 예시 확인</summary>

> 9명 환자의 평균 LDL 콜레스테롤은 121.0mg/dL이지만, 중앙값은 105mg/dL로 평균보다 훨씬 낮습니다.
> 이는 Q009 환자의 매우 높은 수치(245mg/dL)가 평균을 크게 끌어올렸기 때문이며, 이상치를 제외하면 평균(105.5)과 중앙값(103.5)이 비슷해지는 것에서도 이를 확인할 수 있습니다.
> 따라서 이 데이터에서는 "환자 대다수의 일반적인 LDL 수준"을 표현할 때 평균보다 중앙값이 더 안정적인 대표값입니다.
> 다만 Q009의 245mg/dL이 측정 오류인지, 실제 심각한 이상지질혈증 소견인지는 이 데이터만으로 단정할 수 없으므로 추가 확인이 필요하며, 임의로 삭제해서는 안 됩니다.
</details>


---
## 🎉 수고하셨습니다!

이 노트북에서는 1일차 D1-01 ~ D1-05까지 배운 개념(관측치·변수, 데이터 유형, 비율, 평균·중앙값, 이상치 영향)을 **새로운 데이터**에 적용해보았습니다.

1일차 본 교안으로 돌아가 D1-06(분산·표준편차)부터 이어서 학습을 진행하세요.
